### PR TITLE
[OCPCLOUD-1441] Add operator manifests

### PR DESCRIPTION
**Pull request by Fedosin**
This PR sets up Deployment yaml and other resources needed to deploy the Cluster Control Plane Machine Set Operator, as we have with other cluster operators.

### CODE REVIEW

File: manifests/0000_31_control-plane-machine-set-operator_03_deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    include.release.openshift.io/single-node-developer: "true"
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      k8s-app: control-plane-machine-set-operator
+  template:
+    metadata:
+      annotations:
+        target.workload.openshift.io/management: '{"effect": "PreferredDuringScheduling"}'
```
Comment (JoelSpeed) [inline manifests/0000_31_control-plane-machine-set-operator_03_deployment.yaml:20]: Why is this running on workload?

File: manifests/0000_31_control-plane-machine-set-operator_03_deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    spec:
+      priorityClassName: system-node-critical
+      serviceAccountName: control-plane-machine-set-operator
+      containers:
+      - name: control-plane-machine-set-operator
+        image: quay.io/openshift/origin-control-plane-machine-set-operator
+        command:
+        - "/manager"
+        args:
+        - "--zap-devel"
```
Comment (JoelSpeed) [inline manifests/0000_31_control-plane-machine-set-operator_03_deployment.yaml:32]: Where did these come from?

File: manifests/0000_31_control-plane-machine-set-operator_03_deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        operator: "Exists"
+        effect: "NoSchedule"
+      - key: "node.kubernetes.io/unreachable"
+        operator: "Exists"
+        effect: "NoExecute"
+        tolerationSeconds: 120
+      - key: "node.kubernetes.io/not-ready"
+        operator: "Exists"
+        effect: "NoExecute"
+        tolerationSeconds: 120
```
Comment (JoelSpeed) [inline manifests/0000_31_control-plane-machine-set-operator_03_deployment.yaml:59]: We don't need these do we?

**Review by JoelSpeed — COMMENTED** [review]
Please make sure to add an appropriate PR description and make sure the content of the PR is tailored to the current operator, currently it seems quite generic and maybe not quite what we want

/hold this has elements that cannot be merged yet, see comments inline (payload and cluster operator status)

File: manifests/0000_31_control-plane-machine-set-operator_03_deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    matchLabels:
+      k8s-app: control-plane-machine-set-operator
+  template:
+    metadata:
+      annotations:
+        target.workload.openshift.io/management: '{"effect": "PreferredDuringScheduling"}'
+      labels:
+        k8s-app: control-plane-machine-set-operator
+    spec:
+      priorityClassName: system-node-critical
```
[the remark this answers is not in the document]
Comment (Fedosin) [reply manifests/0000_31_control-plane-machine-set-operator_03_deployment.yaml]: Should be `system-cluster-critical`

File: manifests/0000_31_control-plane-machine-set-operator_01_rbac.yaml [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+  annotations:
+    include.release.openshift.io/self-managed-high-availability: "true"
+rules:
+  - apiGroups:
+      - machine.openshift.io
+    resources:
+      - machines
+    verbs:
+      - list
+      - watch
```
Comment (JoelSpeed) [inline manifests/0000_31_control-plane-machine-set-operator_01_rbac.yaml:24]: For Machines, we need the full suite as we will be responsible for creating, updating and deleting the machines
```suggestion
      - get
      - list
      - watch
      - create
      - update
      - patch
      - delete
```

File: manifests/0000_31_control-plane-machine-set-operator_01_rbac.yaml [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+  - apiGroups:
+      - ""
+    resources:
+      - services
+    verbs:
+      - create
+      - watch
+      - get
+      - list
+      - patch
```
Comment (JoelSpeed) [inline manifests/0000_31_control-plane-machine-set-operator_01_rbac.yaml:56]: Are you sure? What's this for?

File: manifests/0000_31_control-plane-machine-set-operator_04_clusteroperator.yaml [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+      version: "0.0.1-snapshot"
+  relatedObjects:
+  - group: ""
+    name: openshift-machine-api
+    resource: namespaces
+  - group: machine.openshift.io
+    name: ""
+    resource: controlplanemachinesets
+  - group: rbac.authorization.k8s.io
+    name: ""
```
[the remark this answers is not in the document]
Comment (Fedosin) [reply manifests/0000_31_control-plane-machine-set-operator_04_clusteroperator.yaml:21]: I removed rbac from relatedObjects. I don't think we really need it here.

File: manifests/0000_31_control-plane-machine-set-operator_03_deployment.yaml (hunk 1 above)
Comment (JoelSpeed) [reply manifests/0000_31_control-plane-machine-set-operator_03_deployment.yaml:19]: You tell me, I don't know what this annotation is 😅  If that's the case, then I think we are good

File: manifests/0000_31_control-plane-machine-set-operator_01_rbac.yaml [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: control-plane-machine-set-operator
+  annotations:
+    include.release.openshift.io/self-managed-high-availability: "true"
+rules:
```
Comment (JoelSpeed) [inline manifests/0000_31_control-plane-machine-set-operator_01_rbac.yaml:49]: We will also need to Get/List/Watch Nodes as this forms part of the Machine Provider logic

**Pull request merged**